Classes/Cache/DatabaseCache.php

```diff
@@ -199,6 +199,15 @@ public function getUrlFromCacheBySpeakingUrl($rootPageId, $speakingUrl, $languag
 				'', 'expire'
 		);
 
+		// More than one urldata entry? Then prefer those with cHash
+		if (count($rows) > 1) {
+			if (preg_match('/cHash/', json_encode($rows))) {
+				$rows = array_filter($rows, function ($row) {
+					return strpos($row['original_url'], 'cHash') !== false ? true : false;
```

Comment:
www.domain.ch/news/contact-us/
www.domain.co.uk/news/contact-us/

URLs are the same but language passed to the function is different because of the _DOMAINS. If you ignore the language, you have 50% chances to find and return the wrong url.